### ISSUE TITLE
Would be possible to open the external building door with ha-sip? 

### ISSUE THREAD

**Issue by marceloaf** [arnonym/ha-plugins#48]
Hello Arne! Hope this finds you well. First of all, thanks for this add-on! 
I'm trying to accomplish the following scenario:

I live in an apartment building and want to be able to open the external door using the add-on you developed. 

- I have a small IP PBX where the intercom is connected. 
- The intercom calls are directed to a SIP extension, which is the extension being monitored by ha-sip.
- When a call comes ha-sip answers the call and present a menu like the one below:

menu:
    timeout: 60
    message: Please enter your access code
    choices_are_pin: true
    choices:
        '1234':
            id: owner
            message: Welcome, Marcelo.
            **# At this moment, I would like to send a DTMF code to open the door**
            post_action: hangup
        'default':
            id: wrong_code
            message: Incorrect code. Try again.
            post_action: return
        'timeout':
            id: timeout
            message: Bye.
            post_action: hangup
            
Note: I inserted the formatted yaml but github didn't honor it. The menu above is configured in sip-1-incoming.yaml and the add-on has the following configuration in sip1. 

enabled: true
registrar_uri: sip:192.168.x.y
id_uri: sip:sip_account@192.168.x.y
realm: "*"
user_name: sip_account
password: extension_password
answer_mode: accept
settle_time: 1
incoming_call_file: /config/sip-1-incoming.yaml

Questions: 
- How could I send a DTMF code only after entering the correct PIN?
- I have already tried to send DTMF codes while a call was established, although I was only successful when the call was originated by ha-sip and not when it was answered.
Any ideas on how to accomplish the scenario above? Any suggestions are very welcome!

**Comment by arnonym** [comment]
Hi @marceloaf,

you can do that with an automation trigger on a webhook. Considering the incoming-sip1.yaml:

```yaml
allowed_numbers:
    - "5555555"
webhook_to_call:
    entered_menu: entered_menu_webhook_id  # this is the important part
menu:
    timeout: 60
    message: Please enter your access code
    choices_are_pin: true
    choices:
        '1234':
            id: owner
            message: Welcome, Marcelo.
            post_action: hangup
        'default':
            id: wrong_code
            message: Incorrect code. Try again.
            post_action: return
        'timeout':
            id: timeout
            message: Bye.
            post_action: hangup
```

You can create the following automation:

```yaml
description: Open door
mode: single
trigger:
  - platform: webhook
    webhook_id: entered_menu_webhook_id
condition: []
action:
  - choose:
      - conditions:
          - condition: template
            value_template: >
              {{ trigger.json.event == "entered_menu" and trigger.json.menu_id
              == "owner" }}
        sequence:
          - service: hassio.addon_stdin
            data_template:
              addon: c7744bff_ha-sip
              input:
                command: send_dtmf
                number: sip:**620@fritz.box
                digits: "123#"
                method: in_band
    default: []

```

The web-hook id from the config must match the one from the trigger. Other than that it is checked if the `entered_menu` event is fired with the correct `menu_id` (in this case "owner"). If both are correct, the stdin_in service is used to send the DTMF digits to your active call.

Hope this works for you!

Arne

**Comment by marceloaf** [comment]
Great!! Thank you very much!! 

I was not considering the second condition "trigger.json.menu_id == "owner"" in my automation so I got confused on how to guarantee the correct code was entered!

Thanks again! I will test it later today and let you know!

BR
Marcelo

**Comment by arnonym** [comment]
Can you proclaim success?

**Comment by marceloaf** [comment]
Hi Arne, good evening! Not yet! I spent the last couple of days troubleshooting another issue with my IP PBX and I could only dedicate to this in the middle of the afternoon. Apparently, everything works as expected except that the dtmf digits are not opening the door. I already tested in_band and rfc2833, which should have worked as it’s the default for my IP PBX. Tomorrow I will test with sip_info. 

If I call from the intercom and answer with any of the SIP extensions I can open the door using the dtmf code. 

Any other idea on how to troubleshoot if the dtmf codes are being correctly sent?

Thanks in advance! 
Marcelo

**Comment by arnonym** [comment]
You could call yourself instead of the door station and would at least hear the inband DTMF tones. Other than that no idea, I don't use outbound DTMF myself.

**Comment by marceloaf** [comment]
Ok. Thanks anyway! I also tested sip_info but didn't work either. During tests I can't hear any DTMF sound

**Comment by marceloaf** [comment]
> You could call yourself instead of the door station and would at least hear the inband DTMF tones. Other than that no idea, I don't use outbound DTMF myself.

Yeah! But my initial intent was to have something for emergencies (lost or robbed keys/phone)

**Comment by arnonym** [comment]
This was only meant for debugging purposes. And you would only be able to hear the inband DTMF tones, as the others are not submitted as audio.

**Comment by arnonym** [comment]
Closed because of inactivity.
